src/main/java/org/sonarlint/intellij/util/SonarLintUtils.java

```diff
@@ -108,12 +115,40 @@ public void run() {
     return false;
   }
 
+  public static boolean shouldAnalyzeAutomatically(@Nullable VirtualFile file, @Nullable Module module) {
+    if(!shouldAnalyze(file, module)) {
+      return false;
+    }
+
+    if(file.getFileType().getDefaultExtension().equalsIgnoreCase("java")) {
+      if(!isSource(file, module)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1066)